Tests/UnitTests/Networking/Backend/BackendGetVirtualCurrenciesTests.swift

```diff
@@ -0,0 +1,291 @@
+        expect(rcCoinVC.description).to(beNil())
+        expect(rcCoinVC.name).to(equal("RC Coin"))
+    }
+
+    func testGetVirtualCurrenciesNoCurrencies() throws {
+        self.httpClient.mock(
+            requestPath: .getVirtualCurrencies(appUserID: Self.userID),
+            response: .init(statusCode: .success,
+                            response: Self.noVirtualCurrenciesResponse,
+                            delay: .milliseconds(10))
+        )
+        let result: Atomic<Result<VirtualCurrenciesResponse, BackendError>?> = nil
+
+        self.virtualCurrenciesAPI.getVirtualCurrencies(
+            appUserID: Self.userID,
+            isAppBackgrounded: false
+        ) { result.value = $0 }
+
+        expect(result.value).toEventuallyNot(beNil())
+
+        let response = try XCTUnwrap(result.value?.value)
+        let virtualCurrencies = try XCTUnwrap(response.virtualCurrencies)
+        expect(virtualCurrencies.count).to(equal(0))
+    }
+
+    // MARK: - Error Handling
+    func testGetVirtualCurrenciesFailSendsError() {
+        self.httpClient.mock(
+            requestPath: .getVirtualCurrencies(appUserID: Self.userID),
+            response: .init(error: .unexpectedResponse(nil))
+        )
+
+        let result = waitUntilValue { completed in
+            self.virtualCurrenciesAPI.getVirtualCurrencies(
+                appUserID: Self.userID,
+                isAppBackgrounded: false,
+                completion: completed
+            )
+        }
+
+        expect(result).to(beFailure())
+    }
+
+    func testGetVirtualCurrenciesNetworkErrorSendsError() {
+        let mockedError: NetworkError = .unexpectedResponse(nil)
+
+        self.httpClient.mock(
+            requestPath: .getVirtualCurrencies(appUserID: Self.userID),
+            response: .init(error: mockedError)
+        )
+
+        let result = waitUntilValue { completed in
+            self.virtualCurrenciesAPI.getVirtualCurrencies(
+                appUserID: Self.userID,
+                isAppBackgrounded: false,
+                completion: completed
+            )
+        }
+
+        expect(result).to(beFailure())
+        expect(result?.error) == .networkError(mockedError)
+    }
+
+    // MARK: - Logging Tests
+    func testRepeatedRequestsLogDebugMessage() {
+        self.httpClient.mock(
+            requestPath: .getVirtualCurrencies(appUserID: Self.userID),
+            response: .init(statusCode: .success,
+                            response: Self.twoVirtualCurrenciesResponse,
+                            delay: .milliseconds(10))
+        )
+        self.virtualCurrenciesAPI.getVirtualCurrencies(
+            appUserID: Self.userID,
+            isAppBackgrounded: false
+        ) { _ in }
+        self.virtualCurrenciesAPI.getVirtualCurrencies(
+            appUserID: Self.userID,
+            isAppBackgrounded: false
+        ) { _ in }
+
+        expect(self.httpClient.calls).toEventually(haveCount(1))
```

Comment:
In this case we cannot avoid the `toEventually` so easily because we want the two requests to happen at the same time. So I think this is OK. Although we could potentially use Async/await and trigger both (with `async let`) and then `await` for the two results. But perhaps it's too much to avoid the `toEventually` 😅